inspirehep/utils/url.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Helpers for handling with http requests and URL handling."""
+
+from __future__ import absolute_import, print_function
+
+from flask import current_app
+
+from inspirehep import __version__
+
+
+def make_user_agent_string(component=""):
```

Comment:
Oh, now I realize! If we have a mutable as a default argument then it is initialized only once, so `component` will be whatever we put there the first time.

So we can't do this. We'd have to pass `component=None`, in fact, since `if component` will evaluate to `False`.